cdap-app-fabric/src/test/java/co/cask/cdap/metadata/FakeFieldLineageReader.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metadata;
+
+import co.cask.cdap.api.lineage.field.EndPoint;
+import co.cask.cdap.data2.metadata.lineage.field.EndPointField;
+import co.cask.cdap.data2.metadata.lineage.field.FieldLineageReader;
+import co.cask.cdap.proto.metadata.lineage.ProgramRunOperations;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Fake implementation of the {@link FieldLineageReader} for testing purpose.
+ */
+public class FakeFieldLineageReader implements FieldLineageReader {
+
+  private final Set<String> fields;
+  private final Set<EndPointField> summary;
+  private final Set<ProgramRunOperations> programRunOperations;
+
+  public FakeFieldLineageReader(Set<String> fields, Set<EndPointField> summary,
+                                Set<ProgramRunOperations> programRunOperations) {
+    this.fields = fields;
+    this.summary = summary;
```

Comment:
better to create unmodifiable copies of the input, in case somebody passes a set in and then modifies it, or tries to modify the result of any of the getters.

Reply:
done.